src/puppetlabs/puppetdb/queue.clj

```diff
@@ -103,6 +106,16 @@
       (utf8-truncate sanitized-certname trunc-length)
       sanitized-certname)))
 
+(defn received+producer-ts [received-ts producer-ts]
```

Comment:
Naming gripe: this looks to me like a function that takes two timestamps and encodes them into a string in a way that preserves characters. Can we call it something about 'encoding' in the name, and have a docstring that describes the encoding?